src/commands/appSettings/filterDownloadAppSettings.ts

```diff
@@ -4,65 +4,48 @@
-    let suppressPrompt: boolean = false;
-    let overwriteSetting: boolean = false;
-    let overwriteSettingsToIgnore: boolean = false;
 
     const addedKeys: string[] = [];
     const updatedKeys: string[] = [];
     const userIgnoredKeys: string[] = [];
     const matchingKeys: string[] = [];
-    const securitySettingsIgnored: string[] = [];
+
     const listOfSettingsToIgnore: string[] = ["AzureWebJobsStorage", "WEBSITE_CONTENTAZUREFILECONNECTIONSTRING", "WEBSITE_CONTENTSHARE"];
+    const options: vscode.QuickPickItem[] = [];
+    destinationSettingsToIgnore.length = 0;
 
-    const yesToAll: vscode.MessageItem = { title: localize('yesToAll', 'Yes to all') };
-    const noToAll: vscode.MessageItem = { title: localize('noToAll', 'No to all') };
+    for (const element of Object.keys(sourceSettings)) {
+        options.push({
+            label: element,
+            picked: !listOfSettingsToIgnore.includes(element)
+        });
+    }
 
-    destinationSettingsToIgnore.length = 0;
+    const result = await context.ui.showQuickPick(options, { placeHolder: 'Select the app settings you would like to download:', canPickMany: true });
+    const userChosenSettings: string[] = result ? result.map(item => item.label) : [];
 
     for (const key of Object.keys(sourceSettings)) {
-        if (listOfSettingsToIgnore.includes(key)) {
-            if (!suppressPrompt) {
-                const message: string = localize('overwriteSettingsToIgnore', 'Setting "{0}" contains secrets. Download?', key);
-                const result: vscode.MessageItem = await context.ui.showWarningMessage(message, { modal: true }, DialogResponses.yes, yesToAll, DialogResponses.no, noToAll);
-                overwriteSettingsToIgnore = result === DialogResponses.yes || result === yesToAll;
-                suppressPrompt = result === yesToAll || result === noToAll;
-            }
-            if (overwriteSettingsToIgnore) {
-                updatedKeys.push(key);
-                destinationSettings[key] = sourceSettings[key];
-            } else {
-                securitySettingsIgnored.push(key);
-                destinationSettings[key] = "_REDACTED_";
-                destinationSettingsToIgnore.push(key);
-            }
-        }
-        else {
-            if (destinationSettings[key] === undefined) { // Have an explicit check for undefined as empty settings should not pass the condition
+        if (userChosenSettings.includes(key)) {
+            if (destinationSettings[key] === undefined) { // Explicit check for undefined as destinationSettings[key] could be empty but valid
                 addedKeys.push(key);
                 destinationSettings[key] = sourceSettings[key];
             } else if (destinationSettings[key] === sourceSettings[key]) {
                 matchingKeys.push(key);
             } else if (sourceSettings[key]) {
```

Comment:
It looks like you could combine this with the first if statement (since they both do `updatedKeys` and `destinationSettings`), unless the ordering of these statements matter

Reply:
which if statement would you like me to combine with the first? I am a bit confused.

Reply:
`if (destinationSettings[key] === undefined || sourceSettings[key])`

Reply:
Oh ok, the reason why I have two separate if statements is because if the destination[key] is undefined then it gets added to addedKeys and if it is overwriting an already existing local value then it is added to updatedKeys.

Reply:
Ah you are correct. I missed that those were 2 different variables. My mistakes